runtime/vm/jvminit.c

```diff
@@ -2843,8 +2843,41 @@ modifyDllLoadTable(J9JavaVM * vm, J9Pool* loadTable, J9VMInitArgs* j9vm_args)
 					return JNI_ERR;
 				}
 			}
+
+			/*
+			* On Linux on Z libj9jit dynamically loads libj9zlib as it is used for AOT method data compression
+			* which is currently only enabled on Z platform. We want to ensure that when the JVM loads libj9jit,
+			* libj9zlib is already loaded. See eclipse/openj9#8561 for more details.
+			*/
+#if (defined(S390) && defined(LINUX))
+			{
+			char zlibDll[EsMaxPath];
+			char *zlibDllDir = zlibDll;
+			UDATA expectedZlibPathLength = 0;
+			UDATA zlibDllLength = 0;
+			UDATA zlibFileHandle = 0;
+			
+			zlibDllLength = strlen(vm->j9libvmDirectory);
+			expectedZlibPathLength = zlibDllLength + (sizeof(DIR_SEPARATOR_STR) - 1) + strlen(J9_ZIP_DLL_NAME) + 1;
+			if (expectedZlibPathLength > EsMaxPath) {
+				zlibDllDir = j9mem_allocate_memory(expectedZlibPathLength, OMRMEM_CATEGORY_VM);
```

Comment:
where does this memory get freed?  Likely needs to be added at the end of the scope